lib/src/getDefinitions.test.ts

```diff
@@ -0,0 +1,118 @@
+import { getDefinitions } from "./getDefinitions";
+
+const mockFetch = vi.fn();
+const mockConsole = {
+  warn: vi.fn(),
+  error: vi.fn(),
+};
+
+describe("getDefinitions", () => {
+  beforeAll(() => {
+    vi.stubGlobal("fetch", mockFetch);
+    vi.stubGlobal("console", mockConsole);
+  });
+  afterAll(() => {
+    vi.unstubAllGlobals();
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    vi.unstubAllEnvs();
+  });
+
+  it("should fetch with default config", () => {
+    mockFetch.mockResolvedValue({ json: () => ({ version: 1, features: [] }) });
+
+    expect(getDefinitions()).resolves.toEqual({
+      version: 1,
+      features: [],
+    });
+
+    expect(mockFetch).toHaveBeenCalledWith(
+      "http://localhost:4242/api/client/features",
+      {
+        headers: {
+          Authorization: "default:development.unleash-insecure-api-token",
+          "Content-Type": "application/json",
+          "UNLEASH-APPNAME": "nextjs",
+          "User-Agent": "nextjs",
+        },
+      }
+    );
+  });
+
+  it("should warn about default config", () => {
+    getDefinitions();
+
+    expect(mockConsole.warn).toHaveBeenCalledWith(
+      "Using fallback Unleash API URL (http://localhost:4242/api).",
+      "Provide a URL or set UNLEASH_SERVER_API_URL environment variable."
+    );
+  });
+
+  it("should show an error when using default token", () => {
+    getDefinitions();
+
+    expect(mockConsole.error).toHaveBeenCalledWith(
+      "Using fallback default token. Pass token or set UNLEASH_SERVER_API_TOKEN environment variable."
```

Comment:
Discussion point: tests that check for exact strings in error messages strike me as very brittle. Is there another way we can check for the same thing?

Reply:
https://github.com/Unleash/unleash-client-nextjs/pull/15/commits/336d9ba76adc345df536ca74e2a30d3a7bfc8694

Reply:
That's definitely better, but it's still brittle because you're checking for a string to match something. Is it enough to check that the console has been called with the right level (warn, error, debug)? Or maybe we could extract the strings into variables, so we know they'll be the same (and if they're not, then that's an actual issue)?

Anyway, I don't think this is a blocker, so I'll leave that up to you.